components/brave_shields/browser/ad_block_filters_provider_manager.cc

```diff
@@ -67,55 +51,36 @@ void AdBlockFiltersProviderManager::RemoveProvider(
 }
 
 std::string AdBlockFiltersProviderManager::GetNameForDebugging() {
-  return "AdBlockCustomFiltersProvider";
+  return "AdBlockFiltersProviderManager";
 }
 
 void AdBlockFiltersProviderManager::OnChanged() {
   NotifyObservers();
 }
 
 // Use LoadDATBufferForEngine instead, for Filter Provider Manager.
-void AdBlockFiltersProviderManager::LoadDATBuffer(
-    base::OnceCallback<void(bool deserialize, const DATFileDataBuffer& dat_buf)>
-        cb) {
+void AdBlockFiltersProviderManager::LoadFilterSet(
+    std::shared_ptr<rust::Box<adblock::FilterSet>> filter_set,
+    base::OnceCallback<void()>) {
   NOTREACHED();
 }
 
-void AdBlockFiltersProviderManager::LoadDATBufferForEngine(
+void AdBlockFiltersProviderManager::LoadFilterSetForEngine(
     bool is_for_default_engine,
-    base::OnceCallback<void(bool deserialize, const DATFileDataBuffer& dat_buf)>
-        cb) {
+    std::shared_ptr<rust::Box<adblock::FilterSet>> filter_set,
+    base::OnceCallback<void()> cb) {
   auto& filters_providers = is_for_default_engine
                                 ? default_engine_filters_providers_
                                 : additional_engine_filters_providers_;
-  const auto collect_and_merge = base::BarrierCallback<DATFileDataBuffer>(
-      filters_providers.size(),
-      base::BindOnce(&AdBlockFiltersProviderManager::FinishCombinating,
-                     weak_factory_.GetWeakPtr(), std::move(cb)));
+  const auto collect_and_merge = base::BarrierClosure(
+      filters_providers.size(), base::BindOnce(std::move(cb)));
```

Comment:
why are you adding another BindOnce to the callback here?

Reply:
good point, fixed